ci/test/mkpipeline.py

```diff
@@ -26,18 +26,40 @@
 from materialize import git
 from pathlib import Path
 from tempfile import TemporaryFile
-from typing import Any, List, Set, Sequence
+from typing import Any, Iterable, List, Set, Sequence
 import os
 import subprocess
 import sys
 import yaml
 
+# These paths contain "CI glue code", i.e., the code that powers CI itself,
+# including this very script! All of CI implicitly depends on this code, so
+# whenever it changes, we ought not trim any jobs from the pipeline in order to
+# exercise as much of the glue code as possible.
+#
+# It's tough to track this code with any sort of fine-grained granularity, so we
+# err on the side of including too much rather than too little. (For example,
+# bin/resync-submodules is not presently used by CI, but it's just not worth
+# trying to capture that.)
+CI_GLUE_GLOBS = ["bin", "ci", "misc/python"]
+
 
 def main() -> int:
+    # Make sure we have an up to date view of main.
+    spawn.runv(["git", "fetch", "origin", "main"])
+
+    # Print out a summary of all changes.
+    os.environ["GIT_PAGER"] = ""
+    spawn.runv(["git", "diff", "--stat", "origin/main..."])
+
```

Comment:
I don't how likely it is to come up in practice, but I think you actually want to diff between HEAD and the merge-base between HEAD and main. I think that as things stand right now you're more likely to be over-agressive in testing too much, including in branches that are forked off before changes to glue code were merged.

Reply:
Oh wait that's what three dots does, nevermind.